packages/opentelemetry-core/src/utils/environment.ts

```diff
@@ -16,33 +16,81 @@
 
 import { LogLevel } from '../common/types';
 
-export type ENVIRONMENT_MAP = { [key: string]: string | number };
+const DEFAULT_LIST_SEPARATOR = ',';
 
 /**
  * Environment interface to define all names
  */
-export interface ENVIRONMENT {
-  OTEL_LOG_LEVEL?: LogLevel;
-  OTEL_NO_PATCH_MODULES?: string;
+
+type ENVIRONMENT_NUMBERS = {
   OTEL_SAMPLING_PROBABILITY?: number;
   OTEL_SPAN_ATTRIBUTE_COUNT_LIMIT?: number;
   OTEL_SPAN_EVENT_COUNT_LIMIT?: number;
   OTEL_SPAN_LINK_COUNT_LIMIT?: number;
-}
+};
 
-const ENVIRONMENT_NUMBERS: Partial<keyof ENVIRONMENT>[] = [
+const ENVIRONMENT_NUMBERS_KEYS: (keyof ENVIRONMENT_NUMBERS)[] = [
```

Comment:
isn't this enough ?
```
const ENVIRONMENT_NUMBERS_KEYS = Object.keys(ENVIRONMENT_NUMBERS);
```

Reply:
Without the `keyof` typecast, the `ENVIRONMENT_NUMBERS_KEYS` var gets the type `string[]`

Reply:
Same as in the other one, `ENVIRONMENT_NUMBERS` is a type, not an object. Potential solution is also the same.